pandas/tests/indexing/test_scalar.py

```diff
@@ -129,38 +129,79 @@ def test_imethods_with_dups(self):
         result = df.iat[2, 0]
         assert result == 2
 
-    def test_at_to_fail(self):
+    def test_series_at_raises_type_error(self):
         # at should not fallback
         # GH 7814
-        s = Series([1, 2, 3], index=list("abc"))
-        result = s.at["a"]
+        # GH#31724 .at should match .loc
+        ser = Series([1, 2, 3], index=list("abc"))
+        result = ser.at["a"]
         assert result == 1
+        result = ser.loc["a"]
+        assert result == 1
+
         msg = (
-            "At based indexing on an non-integer index can only have "
-            "non-integer indexers"
+            "cannot do label indexing on <class 'pandas.core.indexes.base.Index'> "
```

Comment:
I have to say that I found the previous error message more readable ..

Reply:
I generally agree, that suggests we should improve the existing messages for "loc"